src/mock_client.c

```diff
@@ -292,9 +292,18 @@ snapshot_thread(void *args)
 		io_wait_time = random() % 2 + 2;
 		wait_time = random() % 2 + 4;
 
+		MTX_LOCK(&spec->rq_mtx);
```

Comment:
comment would be good to understand why we are doing this when we revisit this code